sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -254,7 +254,9 @@ object RewriteDistinctAggregates extends Rule[LogicalPlan] {
 
       // Setup unique distinct aggregate children.
       val distinctAggChildren = distinctAggGroups.keySet.flatten.toSeq.distinct
```

Comment:
not sure if this is necessary, but it's better to use `ExpressionSet(distinctAggGroups.keySet.flatten).toSeq`, instead of calling `.distinct` on `Seq[Expression]`